src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheck.java

```diff
@@ -135,7 +135,7 @@ public boolean isCommentNodesRequired() {
 
     @Override
     public void beginTree(DetailAST rootAST) {
-        if (xpathExpression != null) {
+        if (!query.isEmpty()) {
```

Comment:
`xpathExpression` depends on `query`, if `query` is empty the `xpathExpression` will always be null.
Updating this condition allows to use default value for `query` with pitest mutation killed.

Reply:
Ok to merge